dist/angular-eonasdan-datetimepicker.js

```diff
@@ -39,22 +43,60 @@
                     };
 
                     var isDateEqual = function (d1, d2) {
-                        return moment.isMoment(d1) && moment.isMoment(d2) && d1.valueOf() === d2.valueOf();
+                      if(moment.isMoment(d1) && moment.isMoment(d2)){
+                        if(d1.isValid() && d2.isValid()){
+                          return (d1.valueOf() === d2.valueOf());
+                        }
+                      } else if ((!d1 && d2 && (d1 === d1)) || (d1 && !d2 && (d2 === d2))){
+                        return false; // the d1 === d1 is to avoid NaN which is passed in iniitilization
+                      }
+                      return true; // return true for invald scenarios to prevent possible infinite digest loop
```

Comment:
what are the _invalid_ scenarios?
and why would you return `true` on `isDateEqual` check when they are **invalid**? 

If properties are not date objects, or whatever is they reason why they can't be checked - I think- this method should return `false`.